modules/virt-accessing-vm-yaml-ssh.adoc

```diff
@@ -0,0 +1,156 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vm-yaml-ssh_{context}"]
+= Accessing a virtual machine via SSH with YAML configurations
+
+You can enable a SSH connection to a virtual machine (VM) without the need to run the `virtctl expose` command. When the YAML file for the VM and the YAML file for the service are configured and applied, the service forwards the SSH traffic to the VM.
+
+The following examples show the configurations for the VM's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* Create a namespace for the VM's YAML file by using the `oc create namespace` command and specifying a name for the namespace.
+
+.Procedure
+. In the YAML file for the VM, add the label and a value for exposing the service for SSH connections. Enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  namespace: ssh-ns
+  labels:
+    special: vm-ssh <1>
+  name: vm-ssh
+spec:
+  running: false
+  template:
+   metadata:
+    labels:
+     kubevirt.io/vm: vm-ssh
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {} <2>
+        name: testmasquerade <3>
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+  volumes:
+  - name: containerdisk
+    containerDisk:
+      image: kubevirt/fedora-cloud-container-disk-demo
+  - name: cloudinitdisk
+    cloudInitNocloud:
+...
```

Comment:
There are some problems with this VM yaml, and it won't create a valid VM:
1. The label should be moved from `metadata.labels` to `spec.template.metadata.labels`. Sorry for this mistake, I missed it in the previous review. Note that this error means that a VM can be created, but no service selector will recognize this label, hence - the attempt to ssh the VM via the service will fail.
2. There are 2 `spec` entries here. The second `spec` (with everything under it) should be shifted to be under `spec.template`.
3. It's missing the `userData` section in `cloudInitNoCloud`. Given that these are only 3 lines, I think it's worth adding them anyway.

To make things easy - I suggest using this VM yaml. i verified its validity E2E, which means I managed to ssh the VM via the service:
```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  namespace: ssh-ns
  name: vm-ssh
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/vm: vm-ssh
        special: vm-ssh
    spec:
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: containerdisk
          - disk:
              bus: virtio
            name: cloudinitdisk
          interfaces:
          - masquerade: {}
            name: testmasquerade
          rng: {}
        machine:
          type: ""
        resources:
          requests:
            memory: 1024M
      networks:
      - name: testmasquerade
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: kubevirt/fedora-cloud-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userData: |
              #!/bin/bash
              echo "fedora" | passwd fedora --stdin
```

Reply:
Updated the content. Please check.